cmd/agent/api/server.go

```diff
@@ -69,124 +77,58 @@ func StartServer(
 	invHost inventoryhost.Component,
 	secretResolver secrets.Component,
 ) error {
-	err := initializeTLS()
+	apiAddr, err := getIPCAddressPort()
 	if err != nil {
-		return fmt.Errorf("unable to initialize TLS: %v", err)
+		return fmt.Errorf("unable to get IPC address and port: %v", err)
 	}
 
-	// get the transport we're going to use under HTTP
-	listener, err = getListener()
-	if err != nil {
-		// we use the listener to handle commands for the Agent, there's
-		// no way we can recover from this error
-		return fmt.Errorf("Unable to create the api server: %v", err)
+	additionalHostIdentities := []string{apiAddr}
+
+	ipcServerHost, ipcServerHostPort, ipcServerEnabled := getIPCServerAddressPort()
+	if ipcServerEnabled {
+		additionalHostIdentities = append(additionalHostIdentities, ipcServerHost)
 	}
 
-	err = util.CreateAndSetAuthToken()
+	tlsKeyPair, tlsCertPool, err := initializeTLS(additionalHostIdentities...)
 	if err != nil {
-		return err
+		return fmt.Errorf("unable to initialize TLS: %v", err)
 	}
 
-	// gRPC server
-	authInterceptor := grpcutil.AuthInterceptor(parseToken)
-	opts := []grpc.ServerOption{
-		grpc.Creds(credentials.NewClientTLSFromCert(tlsCertPool, tlsAddr)),
-		grpc.StreamInterceptor(grpc_auth.StreamServerInterceptor(authInterceptor)),
-		grpc.UnaryInterceptor(grpc_auth.UnaryServerInterceptor(authInterceptor)),
+	tlsConfig := &tls.Config{
+		Certificates: []tls.Certificate{*tlsKeyPair},
+		NextProtos:   []string{"h2"},
+		MinVersion:   tls.VersionTLS12,
 	}
 
-	s := grpc.NewServer(opts...)
-	pb.RegisterAgentServer(s, &server{})
-	pb.RegisterAgentSecureServer(s, &serverSecure{
-		configService: configService,
-		taggerServer:  taggerserver.NewServer(tagger.GetDefaultTagger()),
-		// TODO(components): decide if workloadmetaServer should be componentized itself
-		workloadmetaServer: workloadmetaServer.NewServer(wmeta),
-		dogstatsdServer:    dogstatsdServer,
-		capture:            capture,
-	})
-
-	dcreds := credentials.NewTLS(&tls.Config{
-		ServerName: tlsAddr,
-		RootCAs:    tlsCertPool,
-	})
-	dopts := []grpc.DialOption{grpc.WithTransportCredentials(dcreds)}
-
-	// starting grpc gateway
-	ctx := context.Background()
-	gwmux := runtime.NewServeMux()
-	err = pb.RegisterAgentHandlerFromEndpoint(
-		ctx, gwmux, tlsAddr, dopts)
-	if err != nil {
-		return fmt.Errorf("error registering agent handler from endpoint %s: %v", tlsAddr, err)
+	if err := util.CreateAndSetAuthToken(); err != nil {
```

Comment:
Do we have any tests around the token validation on either the old server code or the new?